protocol/vm/vmcontext.go

```diff
@@ -0,0 +1,27 @@
+package vm
+
+type VMContext interface {
```

Comment:
> The name vm.VMContext stutters, it should prob be vm.Context IMO.

Ha, I had it that way first but thought better of it when I considered the potential for confusion. Your point about `vm.Context` is well-taken, though.

> This type might work better as a struct with optional fields

[I tried it](https://github.com/chain/chain/commit/df6946c7e2d255222758ee2b720dbe3ccf1656b8) and expected to hate it, but it's a definite improvement. Thanks!